app/scripts/ui.js

```diff
@@ -99,6 +103,16 @@ async function start() {
     const method = message?.data?.method;
 
     if (method !== METHOD_START_UI_SYNC) {
```

Comment:
Hmm. Presumably we should be showing an error for _any_ initialization error. Maybe I'm missing somewhere we're already doing that.

Either way, can handle in a later separate PR

Reply:
I started going down that route before realizing it required a bit more thought and generalization before implementing it.

Created an issue for it here https://github.com/MetaMask/metamask-extension/issues/32111